src/relaxations/sparse/Chordal.jl

```diff
@@ -1,76 +1,58 @@
+using CliqueTrees, Graphs
+using CliqueTrees: EliminationAlgorithm
+
 """
-    chordal_completion!(G::Graphs.SimpleGraph)
+    chordal_completion!(G::Graphs.SimpleGraph; alg::EliminationAlgorithm=MF())
 
 Augment `G` by a chordal completion using a greedy minimal fill-in, and also return a perfect elimination ordering.
 This is a more efficient implementation of `ChordalGraph.jl/GreedyOrder` (for the "MF" case).
 
 See also [`ChordalGraph.jl`](https://github.com/wangjie212/ChordalGraph)
 
 """
-function chordal_completion!(G::Graphs.SimpleGraph)
-    n = Graphs.nv(G)
-    H = copy(G)
-    order = zeros(Graphs.eltype(G), n)
-    @inbounds for i = 1:n
-        fill_val = typemax(Int)
-        fill_pos = 1
-        for j = 1:n
-            neib = Graphs.neighbors(H, j)
-            lneib = length(neib)
-            if lneib > 0
-                sg = 0
-                for k = 1:lneib, l = k+1:lneib
-                    if Graphs.has_edge(H, neib[k], neib[l])
-                        sg += 1
-                    end
-                end
-                new_fill_val = ((lneib * (lneib - 1)) >> 1) - sg
-                if new_fill_val < fill_val
-                    fill_val = new_fill_val
-                    fill_pos = j
-                end
-            elseif order[j] == 0 && fill_val > 0
-                fill_val = 0
-                fill_pos = j
-                break
-            end
-        end
-        order[fill_pos] = i
-        neib = copy(Graphs.neighbors(H, fill_pos))
-        for j = 1:length(neib)
-            Graphs.rem_edge!(H, fill_pos, neib[j])
-            for k = j+1:length(neib)
-                if Graphs.add_edge!(H, neib[j], neib[k])
-                    Graphs.add_edge!(G, neib[j], neib[k])
-                end
-            end
+function chordal_completion!(graph::SimpleGraph; alg::EliminationAlgorithm=MF())
+    # compute a tree decomposition using
+    # the given algorithm
+    order, tree = cliquetree(graph; alg)
+    filledgraph = FilledGraph(tree)
+    
+    # triangulate graph
+    for v in vertices(filledgraph)
+        for w in neighbors(filledgraph, v)
+            add_edge!(graph, order[v], order[w])
         end
     end
-    return G, order
+    
+    return graph, order
 end
```

Comment:
The whole `chordal_completion!` function is now no longer necessary at all. It was just a helper; only `chordal_cliques!` is ever used. So you can remove it entirely (and the docstring in `docs/src/auxreference.md`).